### PR TITLE
matrix-sliding-sync: init at 0.99.3

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    repo = "sliding-sync";
+    rev = "v${version}";
+    hash = "sha256-lmmOq0gkvrIXQmy3rbTga0cC85t0LWjDOqrH1NWUpdA=";
+  };
+
+  vendorHash = "sha256-447P2TbBUEHmHubHiiZCrFVCj2/tmEuYFzLo27UyCk4=";
+
+  postInstall = ''
+    rm $out/bin/testmigration
+  '';
```
Comment (mweinelt) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:21]: Lacks a reason for the removal.

**Comment by mweinelt** [comment]
And the commit message should say `init`, not `ini`.

**Comment by K900** [comment]
Would be nice to have a NixOS module to set this up correctly too. I had a small local setup with https://gitlab.com/K900/nix/-/blob/master/machines/konata/synapse-sliding-sync.nix, could be a starting point I guess?

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix (hunk 1 above)
Comment (emilylange) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:21]:
```suggestion
  subPackages = [ "cmd/syncv3" ];
```

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
```
Comment (emilylange) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:22]:
```suggestion

  ldflags = [
    "-X main.GitCommit=${src.rev}"
  ];

```

Fixed version reporting.

Before:
```
❯ result/bin/syncv3
Sync v3 [0.99.3] ()
```

After:
```
❯ result/bin/syncv3
Sync v3 [0.99.3] (v0.99.3)
```

https://github.com/matrix-org/sliding-sync/blob/57c25abc802e8b08f8662dce7791f60b00228868/Dockerfile#L11
https://github.com/matrix-org/sliding-sync/blob/57c25abc802e8b08f8662dce7791f60b00228868/cmd/syncv3/main.go#L23

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  # requires a running matrix-synapse
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Run a sliding sync proxy. An implementation of MSC3575";
```
Comment (emilylange) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:27]:
```suggestion
    description = "A matrix.org sliding sync implementation of MSC3575";
```

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    homepage = "https://github.com/matrix-org/sliding-sync";
+    license = with licenses; [ asl20 ];
+    maintainers = with maintainers; [ SuperSandro2000 ];
```
Comment (emilylange) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:30]:
```suggestion
    maintainers = with maintainers; [ SuperSandro2000 emilylange ];
```

If you don't mind

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
> Would be nice to have a NixOS module to set this up correctly too. I had a small local setup with [gitlab.com/K900/nix/-/blob/master/machines/konata/synapse-sliding-sync.nix](https://gitlab.com/K900/nix/-/blob/master/machines/konata/synapse-sliding-sync.nix), could be a starting point I guess?

Remind me in 2 weeks when I got it working and tested

image

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    rm $out/bin/testmigration
+  '';
+
+  # requires a running matrix-synapse
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Run a sliding sync proxy. An implementation of MSC3575";
+    homepage = "https://github.com/matrix-org/sliding-sync";
+    license = with licenses; [ asl20 ];
```
Comment (emilylange) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:30]:
```suggestion
    license = licenses.asl20;
    mainProgram = "syncv3";
```

**Review by emilylange — CHANGES_REQUESTED** [review]

**Review by emilylange — APPROVED** [review]

**Pull request merged**